### PR TITLE
Fix issue in toString() for decimal zero

### LINKED ISSUE #34472: Incorrect output of `toString` for decimal with tailing zeros
**Description:**
```
import ballerina/io;
public function main() returns error? {
    decimal d1 = 0.0d;
    decimal d2 = 0d;
    io:println(check decimal:fromString(d1.toString()) === check decimal:fromString(d2.toString())); // should be false
}
```
When there are tailing zeros after decimal point, `toString` does not consider them.

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
Fix issue in toString() for decimal zero with different precisions

Fixes #34472

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by KavinduZoysa — APPROVED** [review]

File: langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal [hunk 1]
```diff
@@ -2278,3 +2259,52 @@ function testUsingCloneableReturnType() {
...
+
+    assertFalse(d1.toString() == d2.toString());
+    assertTrue(check decimal:fromString(d1.toString()) == check decimal:fromString(d2.toString()));
+    assertFalse(decimal:fromString(d1.toString()) === decimal:fromString(d2.toString()));
+
+    d1 = 0.00000000d;
+    d2 = 0.0d;
+
+    assertFalse(d1.toString() == d2.toString());
+    assertTrue(check decimal:fromString(d1.toString()) == check decimal:fromString(d2.toString()));
```
Comment (MaryamZi) [inline langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal:2274]: If `check decimal:fromString(d1.toString())` results in the error being returned from this function, the test will not fail. IMO, it's better to handle the error here or use checkpanic.

Comment (rdulmina) [reply langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal]: replaced  with `checkpanic` thanx

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/DecimalValue.java [hunk 2]
```diff
@@ -253,7 +253,7 @@ public Object frozenCopy(Map<Object, Object> refs) {
      * @param parent The link to the parent node
      */
     public String stringValue(BLink parent) {
-        if (this.valueKind != DecimalValueKind.OTHER) {
+        if (this.valueKind != DecimalValueKind.OTHER && this.valueKind != DecimalValueKind.ZERO) {
```
Comment (chiranSachintha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/DecimalValue.java:256]: Consider these three cases,
**Case 1**
```ballerina
public function main() {
   decimal d1 = 0.0;
   decimal d2 = 0E-1;
   assertEquality(d1.toString(), d2.toString()); // true
}
```
**Case 2**

```ballerina
public function main() {
   decimal d1 = 1000;
   decimal d2 = 1E3;
   assertEquality(d1.toString(), d2.toString()); // false
}
```
**Case 3**
```ballerina
public function main() {
   decimal d1 = 0.1;
   decimal d2 = 1E-1;
   assertEquality(d1.toString(), d2.toString()); // true
}
```
Is the output of first and third cases correct?  According to the spec result of `ToString()` should be in the expression style(not considering the value). Isn't it? @MaryamZi @KavinduZoysa

Comment (rdulmina) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/DecimalValue.java:256]: As per the offline discussion created separate issue to address this #34648

**Review by chiranSachintha — APPROVED** [review]

**Review by KavinduZoysa — APPROVED** [review]

**Pull request merged**